src/test/java/online/carsharing/service/impl/RentalServiceImplTest.java

```diff
@@ -0,0 +1,341 @@
+import online.carsharing.dto.request.rental.ActualReturnDateDto;
+import online.carsharing.dto.request.rental.RentalRequestDto;
+import online.carsharing.dto.response.car.CarResponseDto;
+import online.carsharing.dto.response.rental.RentalResponseDto;
+import online.carsharing.entity.Car;
+import online.carsharing.entity.Rental;
+import online.carsharing.entity.Role;
+import online.carsharing.entity.RoleType;
+import online.carsharing.entity.User;
+import online.carsharing.exception.CarIsNotAvailableException;
+import online.carsharing.exception.EntityNotFoundException;
+import online.carsharing.exception.InvalidInputDataException;
+import online.carsharing.exception.UnauthorizedAccessException;
+import online.carsharing.mapper.RentalMapper;
+import online.carsharing.repository.car.CarRepository;
+import online.carsharing.repository.rental.RentalRepository;
+import online.carsharing.repository.user.UserRepository;
+import online.carsharing.service.NotificationService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class RentalServiceImplTest {
+    private static final int ACTIVE_RENTALS_AMOUNT = 3;
+    private static final LocalDate RENTAL_DATE = LocalDate.now().minusDays(ACTIVE_RENTALS_AMOUNT);
+    private static final int INVENTORY_ADJASTMENT = 1;
+    private static final LocalDate RETURN_DATE = LocalDate.now().plusDays(INVENTORY_ADJASTMENT);
+    private static final long CAR_ID = 1L;
+    private static final long MANAGER_ID = 1L;
+    private static final long CUSTOMER_ID = 2L;
+    private static final long RENTAL_ID = 1L;
+    private static final int INACTIVE_RENTALS_AMOUNT = 5;
+    private static final long NON_EXISTENT_RENTAL_ID = 100L;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RentalRepository rentalRepository;
+
+    @Mock
+    private CarRepository carRepository;
+
+    @Mock
+    private RentalMapper rentalMapper;
+
+    @Mock
+    private NotificationService notificationService;
+
+    @InjectMocks
+    private RentalServiceImpl rentalService;
+
+    private Rental validRental;
+    private User manager;
+    private User customer;
+    private Car car;
+
+    @BeforeEach
+    public void setUp() {
+        Role customerRole = new Role();
+        customerRole.setType(RoleType.CUSTOMER);
+
+        Role managerRole = new Role();
+        managerRole.setType(RoleType.MANAGER);
+
+        manager = new User();
+        manager.setId(MANAGER_ID);
+        manager.setRoles(Set.of(managerRole));
+
+        customer = new User();
+        customer.setId(CUSTOMER_ID);
+        customer.setRoles(Set.of(customerRole));
+
+        car = new Car();
+        car.setId(CAR_ID);
+        car.setInventory(2);
```

Comment:
```suggestion
        car.setInventory(INVENTORY_AMOUNT);
```
![200w](https://github.com/jv-feb24-team1/car-sharing-service/assets/73792778/a757546f-594c-4afb-9c3c-73de4a9f0ad7)